pallets/foreign-investments/src/impls.rs

```diff
@@ -209,23 +213,64 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 	}
 }
 
+impl<T: Config> Pallet<T> {
+	fn deposit_apply_swap_events(
+		who: &T::AccountId,
+		swap_id: SwapId<T>,
+		swap: &SwapOf<T>,
+		status: &SwapStatus<T::SwapBalance, T::SwapRatio>,
+	) {
+		if !status.swapped.is_zero() {
+			Pallet::<T>::deposit_event(Event::SwapFullfilled {
+				who: who.clone(),
+				swap_id,
+				remaining: Swap {
+					amount_out: status.pending,
+					..swap.clone()
+				},
```

Comment:
Note that `..swap.clone()` contains the `currency_in` and `currency_out` to know the swap direction.